### PR TITLE
PLAT-75346: Update design of moonstone/Header with Input

### PR DESCRIPTION
**Pull request by ryanjduffy**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [X] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed

* [X] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
`moonstone/Header` needs to be updated to reflect the new designs when using a `headerInput`

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
* Expose the `focused` class on `moonstone/Input` to adjust the necessary colors from `Header`
* Update the CSS in `Header` for the new designs

### Additional Considerations
* Changed the `.title` margin to 2px which, though non-standard and not divisible by 3, allows the input and title text to match in all resolutions except HD (where 3px does the trick). Would like to find a better alternative to this approach.

### CODE REVIEW

File: packages/moonstone/Panels/Header.module.less [hunk 1]
```diff
@@ -113,24 +113,30 @@
 
 		&.standard {
 			.decorator {
-				background-color: transparent;
+				&:not(.focused) {
```
Comment (Djspaceg) [inline packages/moonstone/Panels/Header.module.less]: Is it possible to do the opposite of this? We're setting background-color, input color, and placeholder color below already in the `focus` state section.

File: packages/moonstone/Panels/Header.module.less [hunk 2]
```diff
@@ -62,12 +62,12 @@
...
 
 		.title {
-			// This `margin-top: -3px;` is intended to account for the height disparity and visual
+			// This `margin-top: 2px;` is intended to account for the height disparity and visual
 			// offset when switching between Header+Input and a Standard Header's title height.
 			// The height is determined by the line-height below, but we want to maintain that as a
 			// relative unit (`em`) rather than dictate specific pixels. The -3px offset may need to
 			// be updated or removed to accomodate taller glyphs.
-			margin-top: -3px;
+			margin-top: 2px;
```
Comment (Djspaceg) [inline packages/moonstone/Panels/Header.module.less]: This is indeed not ideal. Things that affect layout really shouldn't have a non-multiple of 3. :\

File: packages/moonstone/Input/Input.js [hunk 3]
```diff
@@ -231,7 +232,7 @@ const InputBase = kind({
 	styles: {
 		css: componentCss,
 		className: 'decorator',
-		publicClassNames: ['decorator', 'input']
+		publicClassNames: ['decorator', 'focused', 'input']
```
Comment (Djspaceg) [inline packages/moonstone/Input/Input.js]: Is this absolutely necessary to achieve the desired style?

**Comment by ryanjduffy** [comment]
Switching to `:focus-within` simplified things quite a bit. Thanks for the suggestion, @Djspaceg!

**Comment by ryanjduffy** [comment]
I've removed the two color variables and updated the change log in which i included a `Removed` entry for the `focused` prop and a generic `Changed` entry for the Header visual updates.

I'll make another PR targeting `develop` for the deprecation notice for `focused`. I doubt I'll be able to safely warn on usage since it's a prop that's actively used by `InputSpotlightDecorator` in that version so it would warn all the time.

File: packages/moonstone/Input/Input.module.less [hunk 4]
```diff
@@ -132,7 +132,7 @@
 				color: @moon-input-placeholder-color;
 			});
 
-			.focused& {
+			&:focus-within {
```
Comment (webOS101) [inline packages/moonstone/Input/Input.module.less]: Is this OK to use?  It's not supported in Blink 53. Do we need a fallback or alternative solution for older browsers?  Or, should we document that 3.x can't be used on older Blink?

Comment (ryanjduffy) [reply packages/moonstone/Input/Input.module.less]: Hmm. Good question. I think we're safe but we should confirm that apps built with 3.0 will not be deployed to older platforms.

Comment (Djspaceg) [reply packages/moonstone/Input/Input.module.less]: I'm curious what other JavaScript and CSS features we already use that are not compatible with such an old browser (released in August 2016).

Comment (ryanjduffy) [reply packages/moonstone/Input/Input.module.less]: We've been pretty careful out of necessity but there is at least one guarded case (e.g. `ResizeObserver`). Not sure how many others though.

**Pull request merged**